include/siminusminus/containers/inmutableiterator.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef SIMINUSMINUS_CONTAINERS_INMUTABLEITERATOR_HPP
+#define SIMINUSMINUS_CONTAINERS_INMUTABLEITERATOR_HPP
+
+#include <algorithm>
+
+namespace cmm {
+namespace containers {
+
+/**
+ * \ingroup containers
+ * \brief iterator to inmutable objects.
+ * 
+ * The InmutableIterator class allows to create an iterator which cannot be
+ * modified, only allows read operations.
+ */
+template <typename T>
+class InmutableIterator : public std::iterator<std::input_iterator_tag, // iterator_category
+                                               const T* const 			// value_type
+                                              > 
+{
+
+public:
+
+	/**
+	 * Default constructor. Init the iterator to null.
+	 */
+	InmutableIterator(): _pointer(nullptr)
+	{
+
+	}
+
+	/**
+	 * Constructor that takes a T object to point to object memory direction.
+	 * @param object: object of T Type to init the iterator.
+	 */
+	InmutableIterator(const T& object) : _pointer(&object)
```

Comment:
I'm not sure this is the expected behavior. I would make passing the address to the sequence element explicit